apps/designer/app/designer/features/props-panel/use-props-logic.ts

```diff
@@ -115,16 +185,20 @@ export const usePropsLogic = ({
       ? []
       : selectedInstanceData.props.props;
 
-  const initialState = uniqBy(
-    [
-      ...props,
-      ...getPropsWithDefaultValue(selectedInstanceData),
-      ...getRequiredProps(selectedInstanceData),
-    ],
-    "prop"
+  const [requiredProps] = useState<Array<UserProp>>(() =>
+    uniqBy(
+      [
+        ...getInitialProps(selectedInstanceData),
+        ...getRequiredProps(selectedInstanceData),
+        ...getPropsWithDefaultValue(selectedInstanceData),
+      ],
```

Comment:
If `getInitialProps` and `getRequiredProps` used only here, will it make sense to override with values from `selectedInstanceData.props` somewhere here rather than in each of these functions?

Reply:
I have feeling that everything in this logic will be changed, hard to say now. For now just prevented that props user added/removed after reload occured or moved